sharktank/tests/models/llama/benchmark_tests.py

```diff
@@ -0,0 +1,779 @@
+            f"--input=@{self.decode_args_fp8}/seq_lens.npy",
+            f"--input=@{self.decode_args_fp8}/start_positions.npy",
+            f"--input=@{self.decode_args_fp8}/seq_block_ids.npy",
+            f"--input=@{self.decode_args_fp8}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+
+    @longrun
+    def testBenchmark8B_f16_Decomposed(self):
+        output_mlir = self.repo_root + "llama8b_f16_decomposed.mlir"
+        output_json = self.repo_root + "llama8b_f16_decomposed.json"
+        output_vmfb = self.repo_root + "llama8b_f16_decomposed.vmfb"
+        self.export_mlir(
+            "decomposed",
+            self.tensor_parallelism_size,
+            self.irpa_path,
+            output_mlir,
+            output_json,
+            self.repo_root,
+        )
+        iree_compile_args = self.iree_compile_args + [
+            f"--iree-hal-dump-executable-files-to={self.repo_root}/files/llama-8b/f16-decomposed"
+        ]
+        self.iree_compile(output_mlir, output_vmfb, iree_compile_args, self.repo_root)
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            output_vmfb,
+            self.irpa_path,
+            self.iree_run_prefill_args,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            output_vmfb,
+            self.irpa_path,
+            self.iree_run_decode_args,
+            self.repo_root,
+        )
+        self.cleanup_output_files(output_mlir, output_json, output_vmfb)
+
+    @pytest.mark.skip(reason="TODO: Need to plumb through attention_kernel")
+    @longrun
+    def testBenchmark8B_f16_Non_Decomposed(self):
+        output_mlir = self.repo_root + "llama8b_f16_torch_sdpa.mlir"
+        output_json = self.repo_root + "llama8b_f16_torch_sdpa.json"
+        output_vmfb = self.repo_root + "llama8b_f16_torch_sdpa.vmfb"
+        self.export_mlir(
+            "torch_sdpa",
+            self.tensor_parallelism_size,
+            self.irpa_path,
+            output_mlir,
+            output_json,
+            self.repo_root,
+        )
+        iree_compile_args = self.iree_compile_args + [
+            f"--iree-hal-dump-executable-files-to={self.repo_root}/files/llama-8b/f16-torch-sdpa"
+        ]
+        self.iree_compile(output_mlir, output_vmfb, iree_compile_args, self.repo_root)
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            output_vmfb,
+            self.irpa_path,
+            self.iree_run_prefill_args,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            output_vmfb,
+            self.irpa_path,
+            self.iree_run_decode_args,
+            self.repo_root,
+        )
+        self.cleanup_output_files(output_mlir, output_json, output_vmfb)
+
+    @pytest.mark.xfail
```

Comment:
nit: would be good to comment why we have the xfails for any newcomers not familiar with what isn't functional yet